orangecontrib/text/tests/test_twitter.py

```diff
@@ -0,0 +1,192 @@
+
+import tweepy
+from datetime import date
+
+from orangecontrib.text import twitter
+from orangecontrib.text.corpus import Corpus
+
+
+def get_credentials():
+    key = os.environ.get('TWITTER_KEY', '')
+    secret = os.environ.get('TWITTER_SECRET', '')
+    if key and secret:
+        return twitter.Credentials(key, secret)
+    return twitter.Credentials('key', 'secret')
+
+
+NO_CREDENTIALS = os.environ.get('TWITTER_KEY', None) is None
+CREDENTIALS_MSG = "No twitter api credentials have been found."
+
+
+class TestCredentials(unittest.TestCase):
+
+    @unittest.skipIf(NO_CREDENTIALS, CREDENTIALS_MSG)
+    def test_valid(self):
+        credentials = get_credentials()
+        self.assertTrue(credentials.valid)
+
+    def test_check_bad(self):
+        key = twitter.Credentials('bad key', 'wrong secret')
+        self.assertFalse(key.valid)
+
+    def test_equal(self):
+        key1 = twitter.Credentials('key1', 'secret1')
+        key2 = twitter.Credentials('key1', 'secret1')
+        self.assertEqual(key1, key2)
+
+        key2.consumer_secret = 'key2'
+        self.assertNotEqual(key1, key2)
+
+        self.assertNotEqual(key1, None)
+
+    def test_pickle(self):
+        key1 = twitter.Credentials('key1', 'secret1')
+        pickled = pickle.dumps(key1)
+        key2 = pickle.loads(pickled)
+        self.assertEqual(key1, key2)
+
+
+class MyCursor:
+    def __init__(self, *args, **kwargs):
+        self.statuses = tweepy.Status.parse_list(
+            None, json.load(open(os.path.join(os.path.dirname(__file__), 'tweets.json'))))
+        self.kwargs = kwargs
+        self.args = args
+
+    def items(self, count):
+        return self.statuses[:count]
+
+
+@mock.patch('tweepy.Cursor', MyCursor)
+class TestTwitterAPI(unittest.TestCase):
+
+    def setUp(self):
+        self.credentials = get_credentials()
+        self.api = twitter.TwitterAPI(self.credentials)
+
+    @unittest.skipIf(NO_CREDENTIALS, CREDENTIALS_MSG)
+    def test_search(self):
+        self.api.search(word_list=['hello'], max_tweets=5, lang='en')
+        self.api.join()
+        self.assertEqual(len(self.api.container), 5)
+        for tweet in self.api.tweets:
+            self.assertIn('hello', tweet['text'].lower())
+            self.assertEqual('en', tweet['lang'])
+
+    def test_search_callbacks(self):
+        self.checker = 0
+
+        def on_start():
+            self.assertEqual(self.checker, 0)
+            self.checker += 1
+
+        def on_progress(progress):
+            self.assertEqual(self.checker, progress)
+            self.checker += 1
+
+        def on_finish():
+            self.assertEqual(self.checker, 3)
+            self.checker += 1
+
+        api = twitter.TwitterAPI(self.credentials, on_start=on_start,
+                                 on_progress=on_progress, on_finish=on_finish)
+        api.search(word_list=['hello'], max_tweets=2, lang='en')
+        api.join()
+        self.assertEqual(self.checker, 4)
+
+        if NO_CREDENTIALS:
```

Comment:
This test also passes without credentials. Can we remove this two lines that skip it?